project/ReflectiveCodeGen.scala

```diff
@@ -20,6 +20,9 @@ object ReflectiveCodeGen extends AutoPlugin {
   val extraGenerators = SettingKey[Seq[String]]("reflectiveGrpcExtraGenerators")
   val codeGeneratorSettings = settingKey[Seq[String]]("Code generator settings")
 
+  // needed to be able to override the PB.generate task reliably
+  override def requires = ProtocPlugin
```

Comment:
I have tried this and the above change but was missing `PB.protoSources.value` below. 

I confess that this is all goes above my head.